src/unity/toolkits/supervised_learning/automatic_model_creation.cpp

```diff
@@ -70,8 +70,8 @@ std::shared_ptr<supervised_learning_model_base> create_automatic_classifier_mode
   }
 
   // Train each model. Save the model and its accuracy.
-  double accuracies[possible_models.size()];
-  std::shared_ptr<supervised_learning_model_base> models[possible_models.size()];
+  double *accuracies = new double[possible_models.size()];
```

Comment:
Let's use std::vector<double>  instead.   Using new like this isn't safe if an exception is thrown between this and the delete, but std::vector handles it correctly.